wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentationFlowActivity.kt

```diff
@@ -0,0 +1,294 @@
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.presentation.PresentationFlowActivity.PromptUnsuccessfulException
+import com.android.identity_credential.wallet.ui.prompt.biometric.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.prompt.consent.showConsentPrompt
+import com.android.identity_credential.wallet.ui.prompt.passphrase.showPassphrasePrompt
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.datetime.Clock
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * A base Activity that provides functions for initiating an MDL Presentation flow and obtaining the
+ * response bytes to send to the requesting party.
+ *
+ * Subclassing Activities call function [showPresentationFlow] with the request bytes parsed to a
+ * [PresentationRequestData]. Response bytes are (eventually) generated after authentication prompts
+ * validate the user successfully, else an exception is thrown.
+ *
+ * Exceptions are thrown for numerous reasons such as,
+ * - [IllegalStateException] if credentials could not be found in the Document; authentication
+ * type was not specified when showing Biometric prompt; a new secure area has not been handled yet
+ * - [PromptUnsuccessfulException] if a prompt is unsuccessful when user cancels the prompt or is
+ * unable to authenticate.
+ */
+open class PresentationFlowActivity : FragmentActivity() {
+    companion object {
+        const val TAG = "PresentationFlowActivity"
+    }
+
+    // reference WalletApplication for obtaining dependencies
+    private val walletApp: WalletApplication by lazy {
+        application as WalletApplication
+    }
+
+    // private class dedicated to managing/providing Mdoc-related functions needed by this Activity
+    private val mdocHelper = MdocHelper()
+
+    /**
+     * Show the Presentation Flow starting with Consent prompt, followed by Biometric prompt
+     * and/or Passphrase prompt. If all the prompts are successful, return the response bytes.
+     * Otherwise throw the Exception that prevented the Presentation flow from generating the
+     * response bytes.
+     *
+     * @param presentationRequestData the request data produced after parsing request bytes
+     * @return the response bytes generated after all prompts succeed.
+     * @throws [IllegalStateException] for incorrect configurations or if cannot find credential.
+     * @throws [PromptUnsuccessfulException] if a prompt does not succeed because the user cancelled
+     * or is unable to authenticate.
+     */
+    suspend fun showPresentationFlow(presentationRequestData: PresentationRequestData): ByteArray {
+        val credential = presentationRequestData.document.getMdocCredential()
+            ?: throw IllegalStateException("Unable to find credential in Document ${presentationRequestData.document.name}")
+
+        // show Consent Prompt
+        showConsentPrompt(
```

Comment:
I don't think that comment is helpful, it's stating the obvious. Generally, try to avoid comments for things that are obvious. In the limit, no comments are needed because all code is so perfectly designed that it's self-explanatory.